.github/workflows/main.yml

```diff
@@ -499,3 +499,127 @@ jobs:
+        uses: actions/checkout@v3
+        with:
+          path: 'repo'
+          ref: ${{ github.ref }}
+
+      - name: 'Download artifact'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        uses: actions/download-artifact@v3
+        with:
+          name: 'obs-studio-macos-${{ matrix.arch }}-notarized'
+          path: 'artifacts'
+
+      - name: 'Install Python requirements'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        run: pip3 install requests xmltodict
+
+      - name: 'Install Brew requirements'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        run: brew install coreutils pandoc
+
+      - name: 'Setup Sparkle'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        run: |
+          wget "https://github.com/sparkle-project/Sparkle/releases/download/${{ env.SPARKLE_VERSION }}/Sparkle-${{ env.SPARKLE_VERSION }}.tar.xz" -O Sparkle.tar.xz
+
+          if [[ '${{ env.SPARKLE_HASH }}' != "$(sha256sum Sparkle.tar.xz | cut -d " " -f 1)" ]]; then
+              echo "Sparkle download hash does not match!"
+              exit 1
+          fi
+
+          mkdir sparkle && cd sparkle
+          tar -xf ../Sparkle.tar.xz
+
+      - name: 'Setup folder structure'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        run: |
+          mkdir builds
+          mkdir -p output/appcasts
+          mkdir -p output/deltas/${{ matrix.arch }}
+
+      - name: 'Determine branch and tag'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        id: branch
+        run: |
+          cd repo/
+          
+          GIT_TAG="$(git describe --tags --abbrev=0)"
+          if [[ ${GIT_TAG} == *'beta'* || ${GIT_TAG} == *'rc'* ]]; then
+            echo "branch=beta" >> $GITHUB_OUTPUT
+            echo "deltas=0" >> $GITHUB_OUTPUT
+          else
+            echo "branch=stable" >> $GITHUB_OUTPUT
+            echo "deltas=1" >> $GITHUB_OUTPUT
+          fi
+          # Write tag description to file
+          git tag -l --format='%(contents)' ${GIT_TAG} >> ../notes.rst
+
+      - name: 'Download existing Appcast and builds'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        env:
+          BRANCH: ${{ steps.branch.outputs.branch }}
+          DELTAS: ${{ steps.branch.outputs.deltas }}
+          ARCH: ${{ matrix.arch }}
+        run: python3 repo/CI/macos/appcast_download.py
+
+      - name: 'Prepare release notes'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        run: |
+          # Insert underline at line 2 to turn first line into heading
+          sed -i '' '2i\'$'\n''###################################################' notes.rst
+          pandoc -f rst -t html notes.rst -o output/appcasts/notes_${{ steps.branch.outputs.branch }}.html
+
+      - name: 'Setup Sparkle key'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        run: echo -n "${{ secrets.SPARKLE_PRIVATE_KEY }}" >> eddsa_private.key
+
+      - name: 'Generate Appcast'
+        if: env.HAVE_SPARKLE_KEY == 'true'
+        run: |
+          mv artifacts/*.dmg builds/
```

Comment:
Ensure that you only handle a single `dmg` file - use a GLOBbing expression that yields only what you need (e.g. store all files sorted by date into an array, then use the first element of the resulting array).

Reply:
In this case we want all downloaded DMG files to be in the builds directory as they are used for delta creation (currently this is limited to 1 previous build).